enhancements/local-storage/automatic-partition.md

```diff
@@ -0,0 +1,193 @@
+- He creates a LocalVolumeSet CR that includes the partitioning policy
+- Once the `LocalVolumeSet` is created, LSO reconciles the LocalVolumeSet CR (the same as designed for full device),
+- The controller will:
+  - configure the local-static-provisioner to make a new StorageClass based on certain directories on the selected nodes.
+  - assign diskmaker daemonset to the selected nodes.
+  - there will be one daemonset for each `LocalVolumeSet` CR.
+- The diskmaker daemon will find devices that match the disovery policy and check if partitoningSpec is present
+  - if partitonSpec is present, it will partition the devices.
+  - as soon as a device is partitoned
+    - the diskmaker will symlink the child partitions of the device into the directory that the local-static-provisioner is watching
+    - The local-static-provisioner will provision PVs on those partitions
+- The localVolumeset won't accept changes after its created and has a partitioningSpec
+
+
+## Design Details for Auto partitioning of local devices
+
+Updated `LocalVolumeSet` API sceme for Auto partition:
+
+
+```go
+
+type LocalVolumeSetSpec struct {
+ ...
+  // PartitioningSpec is the partitioning policy according to which the selected devices will be partitioned.
+  PartitioningSpec PartitoiningSpec `json:"partitioningSpec"`
+}
+
+type PartitioningSpec struct {
+  // PartSize determines the exact size of each partition.
+  PartSize *resource.Quantity `json:"partSize"`
+}
+
+type LocalVolumeSetStatus struct {
+  // Conditions is a list of conditions and their status.
+  Conditions []operatorv1.OperatorCondition `json:"conditions,omitempty"`
+  
+  // TotalProvisionedDeviceCount is the count of the total devices over which the PVs has been provisioned
+  TotalProvisionedDeviceCount *int32 `json:"totalProvisionedDeviceCount,omitempty"`
+  
+  // observedGeneration is the last generation change the operator has dealt with
+  // +optional
+  ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+  
+  // TotalProvisionedPartitionCount is the count of the total partitions over which the PVs has been provisioned
+  // This will be diffrent for TotalProvisionedDeviceCount as TotalProvisionedDeviceCount is
+  //the count of the actual devices
+  TotalProvisionedPartitionCount *int32 `json:"totalProvisionedPartitionCount,omitempty"`
+}
+```
+
+
+### Example of LocalVolumeSet CR:
+
+```yaml
+apiVersion: local.storage.openshift.io/v1alpha1
+kind: LocalVolumeSet
+metadata:
+  name: example-autodetect
+spec:
+  nodeSelector:
+    nodeSelectorTerms:
+      - matchExpressions:
+          - key: kubernetes.io/hostname
+            operator: In
+            values:
+              - worker-0
+              - worker-1
+  storageClassName: example-storageclass
+  volumeMode: Block
+  maxDeviceCount: 10
+  deviceInclusionSpec:
+    deviceTypes:
+      - RawDisk
+    deviceMechanicalProperty:
+      - Rotational
+      - NonRotational
+    minSize: 10G
+    maxSize: 100G
```

Comment:
`maxSize ` must be <= `PartitioningSpec.Size`, good for an internal check.

Reply:
`maxSize` must be >= `PartitioningSpec.Size` right?

Reply:
Right my bad!